workshops/util.py

```diff
@@ -46,22 +43,9 @@ def upload_person_task_csv(uploaded_file, encoding=None):
 
     Also return a list of fields from Person.PERSON_UPLOAD_FIELDS for which
     no data was given.
-
-    :param string encoding: encoding used to encode incoming file. Defaults to
-                            ``django.conf.settings.DEFAULT_CHARSET``
     """
     persons_tasks = []
 
-    if not encoding:
-        encoding = settings.DEFAULT_CHARSET
-
-    # we provide uploaded_file as StringIO in our tests (test_util.py)
-    if not issubclass(StringIO, uploaded_file.__class__):
-        # Django provides uploaded files as byte file objects.  We need to wrap
-        # `uploaded_file` and provide it as a text file with specific encoding.
-        # This way we force users to upload UTF-8 encoded files.
-        uploaded_file = TextIOWrapper(uploaded_file.file, encoding=encoding)
-
     reader = csv.DictReader(uploaded_file)
```

Comment:
On Thu, Feb 05, 2015 at 01:45:14PM -0800, Piotr Banaszkiewicz wrote:
> > -    # we provide uploaded_file as StringIO in our tests (test_util.py)
> > -    if not issubclass(StringIO, uploaded_file.__class__):
> > -        # Django provides uploaded files as byte file objects.  We need to wrap
> > -        # `uploaded_file` and provide it as a text file with specific encoding.
> > -        # This way we force users to upload UTF-8 encoded files.
> > -        uploaded_file = TextIOWrapper(uploaded_file.file, encoding=encoding)
> > -
> >      reader = csv.DictReader(uploaded_file)
> 
> That conditional was required for tests (see for example
> test_util.py:104). Without this conditional, the test suite doesn't
> pass…

They pass for me:

  $ git show HEAD | head -n1
  commit 2545157878bf93c9b5bddb02e838546702a94e10
  $ git status
  On branch drop-py27-changes
  Your branch is up-to-date with 'wking/drop-py27-changes'.
  Untracked files:
    (use "git add <file>..." to include in what will be committed)
  
          amy/whoosh_index/
  
  nothing added to commit but untracked files present (use "git add" to track)
  $ python3.4 manage.py test
  …
  Ran 43 tests in 2.613s

  OK
  Destroying test database for alias 'default'...